clippy_lints/src/formatting.rs

```diff
@@ -127,6 +149,28 @@ fn check_else_if(cx: &EarlyContext, expr: &ast::Expr) {
     }
 }
 
+/// Implementation of the `POSSIBLE_MISSING_COMMA_IN_ARRAY_FORMATTING` lint for array
+fn check_array(cx: &EarlyContext, expr: &ast::Expr) {
+    if let ast::ExprKind::Array(ref array) = expr.node {
+        for element in array {
+            if let ast::ExprKind::Binary(ref op, ref lhs, _) = element.node {
+                let space_span = mk_sp(lhs.span.hi, op.span.lo);
```

Comment:
I see your point. So am I allowed to simply ignore this case (no lint generated) or I need to get the correct span for this (how?).

I ask since I noticed that no `suspicious_assignment_formatting` lint will be generated for this code:
```
macro_rules! foo {
    () => { 42 };
    ($e:expr) => { $e };
}
let mut a = 1;
foo!(a) =- 5; // no lint
a =- 5; // lint
```